### PR TITLE
/cmds/core/uptime: increase test coverage

### PR DESCRIPTION
**Pull request by binjip978**

### CODE REVIEW

File: cmds/core/uptime/uptime.go [hunk 1]
```diff
@@ -11,26 +11,32 @@ package main
...
 	"fmt"
+	"io"
 	"log"
 	"os"
 	"strings"
 	"time"
 )
 
+var (
+	errAverageFormat = errors.New("the contents of proc/loadavg we are trying to process contain less than the required 3 loadavgs")
```
Comment (rminnich) [inline cmds/core/uptime/uptime.go:22]: This is a bit wordy, can we just do something like
errAverageFormat = fmt.Errorf("/proc/loadavg has less than 3 fields:%v", os.ErrInvalid)
errUptimeFormat("/proc/uptime is empty: %v", os.ErrInvalid)

or some such, The os.ErrInvalid is for tests.

File: cmds/core/uptime/uptime.go [hunk 2]
```diff
@@ -11,26 +11,32 @@ package main
...
+
 // loadavg takes in the contents of proc/loadavg,it then extracts and returns the three load averages as a string
 func loadavg(contents string) (loadaverage string, err error) {
 	loadavg := strings.Fields(contents)
 	if len(loadavg) < 3 {
-		return "", fmt.Errorf("error:invalid contents:the contents of proc/loadavg we are trying to process contain less than the required 3 loadavgs")
+		return "", errAverageFormat
 	}
-	return loadavg[0] + "," + loadavg[1] + "," + loadavg[2], nil
+	return loadavg[0] + ", " + loadavg[1] + ", " + loadavg[2], nil
```
Comment (rminnich) [inline cmds/core/uptime/uptime.go:32]: could you use strings.Join here?

File: cmds/core/uptime/uptime.go [hunk 3]
```diff
@@ -11,26 +11,32 @@ package main
...
+	errAverageFormat = errors.New("the contents of proc/loadavg we are trying to process contain less than the required 3 loadavgs")
+	errUptimeFormat  = errors.New("the contents of proc/uptime we are trying to read are empty")
+)
+
 // loadavg takes in the contents of proc/loadavg,it then extracts and returns the three load averages as a string
 func loadavg(contents string) (loadaverage string, err error) {
 	loadavg := strings.Fields(contents)
 	if len(loadavg) < 3 {
-		return "", fmt.Errorf("error:invalid contents:the contents of proc/loadavg we are trying to process contain less than the required 3 loadavgs")
+		return "", errAverageFormat
```
Comment (rminnich) [inline cmds/core/uptime/uptime.go:30]: a log of times, in logfiles, it's nice to see the data we think is defective
return "", fmt.Errorf("%q:%v", contents, errAverageFormat)

File: cmds/core/uptime/uptime.go [hunk 4]
```diff
@@ -41,23 +47,33 @@ func uptime(contents string) (*time.Time, error) {
...
 	if err != nil {
-		log.Fatal(err)
+		return err
 	}
-	procLoadAvgOutput, err := os.ReadFile("/proc/loadavg")
+
+	procLoadAvgOutput, err := os.ReadFile(loadavgPath)
 	if err != nil {
-		log.Fatalf("error reading /proc/loadavg: %v \n", err)
+		return fmt.Errorf("error reading /proc/loadavg: %w", err)
```
Comment (rminnich) [inline cmds/core/uptime/uptime.go:63]: same here, I bet you can return the err all by itself

**Review by rminnich — CHANGES_REQUESTED** [review]

**Comment by rminnich** [comment]
I think with a tiny bit more work the test could be table driven ...

**Comment by binjip978** [comment]
> I think with a tiny bit more work the test could be table driven ...

I thought about that, but one error (time parsing error) making table driven test a bit awkward.

File: cmds/core/uptime/uptime.go (hunk 1 above)
Comment (binjip978) [reply cmds/core/uptime/uptime.go]: We already have errUptimeFormat for test so I think os.ErrInvalid is not needed.
